### PR TITLE
[SYCL][ESIMD] Add ESIMD printf test

### PR DESCRIPTION
**Pull request by fveselov**

### CODE REVIEW

File: SYCL/ESIMD/printf.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+  }
+
+  // FIXME: strictly check output order once the bug mentioned above is fixed
```
Comment (DenisBakhvalov) [inline SYCL/ESIMD/printf.cpp:118]: Is it specified in which order the lines should appear?

Comment (fveselov) [reply SYCL/ESIMD/printf.cpp:118]: This FIXME is related only to printf without %-specifiers, that is the first case in test.
> // Due to a bug in Intel CPU Runtime for OpenCL on Windows, information
   // printed using such format strings (without %-specifiers) might
   // appear in different order if output is redirected to a file or
   // another app

The idea is to remove last CHECK entirely and modify first one to check output order.
Namely remove brackets and question mark, so line 64 would look like:
> // CHECK: Hello, World!

Comment (DenisBakhvalov) [reply SYCL/ESIMD/printf.cpp:118]: I see, That's OK.
Actually, I was asking about CHECKs starting on line 106 whether their order is specified?

Comment (fveselov) [reply SYCL/ESIMD/printf.cpp:118]: No, in current configuration order is not specified.

**Comment by fveselov** [comment]
pre-ci-windows failed on test unrelated to changes

> [2021-08-04T07:24:09.203Z] 3 warning(s) in tests
[2021-08-04T07:24:09.203Z] ********************
[2021-08-04T07:24:09.203Z] Failed Tests (1):
[2021-08-04T07:24:09.203Z]   SYCL :: ESIMD/regression/replicate_bug.cpp

**Review by DenisBakhvalov — APPROVED** [review]

**Comment by DenisBakhvalov** [comment]
@kbobrovs is on vacation.
@vladimirlaz, please merge.

**Pull request merged**